### DISCUSSION TITLE
[Question]: Adjusting Preview GIF Quality

### DISCUSSION THREAD

**Discussion by droans** [blakeblackshear/frigate#12926] [Ask A Question] [answered]
### What is your question:

Is it possible to adjust the quality of the GIF returned by the `/api/events/<event-id>/preview.gif` endpoint? 

It looks like it could be very useful for notifications but I'd prefer if I could increase the quality.

 
![preview](https://github.com/user-attachments/assets/a0b94866-d4d6-4888-9635-78a34e5bd455)

**Comment by NickM-27** [comment, marked as answer]
you can increase the quality of the previews themselves in the config, but there is no way to adjust the compression that is done for the gif. However, this seems like even in its current form would work fine on a phone as a notification

**Reply by droans** [reply]
For sure it's usable. The compression does become more obvious on a high-density display, though.

You are right, I should have read the code first - [the settings are hard-coded](https://github.com/blakeblackshear/frigate/blob/99e03576bfd6b473db9fb368d136453c10b2c528/frigate/api/media.py#L1126-L1148). However, it doesn't look like anything the code is doing can't be done using the API so there's nothing that's stopping me from implementing my own method. 

Thanks!

**Comment by goldserve** [comment]
Possible to use palettegen and paletteuse to output a higher quality gif? https://www.bannerbear.com/blog/how-to-make-a-gif-from-a-video-using-ffmpeg/
